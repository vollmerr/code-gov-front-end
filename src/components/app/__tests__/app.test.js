import React from 'react';

import App from 'components/app'

import { shallow } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <App /> component', () => {
  const props = {
    history: {
      location: {
        hash: "",
        key: "test-key",
        pathname: "/test",
        search: "",
        state: undefined
      },
    },
    location: {
      hash: "",
      key: "test-key",
      pathname: "/test",
      search: "",
      state: undefined
    },
    match: {
      isExact: false,
      params: {},
      path: "/",
      url: "/"
    },
    plugins: [{
      route: "/test-route",
      component: "test/component",
      filename: "test-filename"
    }],
    rehydrate: jest.fn()
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    window.history.pushState({}, 'Test Title', '/test');
    wrapper = shallow(<App />);
  });
  it('renders App', () => {
    expect(wrapper.exists()).toEqual(true);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})