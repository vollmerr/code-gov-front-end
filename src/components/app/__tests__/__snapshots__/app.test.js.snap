// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App /> component should render correctly 1`] = `
<Route>
  <Component />
</Route>
`;
